### DISCUSSION TITLE
Should we increase price / difficulty / whatever? (beginner vs expert player)

### DISCUSSION THREAD

**Discussion by Urban-Hacker** [ochadenas/cpudefense#130] [Ideas]
I see a lot of tickets / discussion about chips being too cheap / too easy and while I’m now an expert player I kinda agree I think we need to keep in mind the slow easy onboarding.

This is one of the thing that made me love this game. It was very easy to get started and before changing all the prices / etc I feel we should not forgot beginners.

I wonder if we could have maybe handicaps like not using shf or mem to achieve certain trophy / reward instead of just making the game harder for everybody. This would also help with re-playbility.

**Comment by Cwpute** [comment]
I feel you're talking about two different things.
About chip prices: the initial Normal levels won't be too much affected by these proposed changes i believe.
Your idea for trophies is an interesting one though that would definitely add a lot of replayability for relatively easy work.

**Comment by Cwpute** [comment]
I came up with a short list of self-imposed handicaps the game could aknowledge and reward the player with a badge:
* Don't use chip X (as you suggested)
* Don't invert attackers by tapping on them (probably easy to mess up this one with a mistap)
* Don't upgrade your chips (probably hard to achieve in every level; maybe something like "don't upgrade above level 3" would be more manageable but still provide a challenge)
* Don't fill every available chip slot
* Don't sell negative chips
* Possess a certain amount of unused bits by the end of the level (this one i don't think is possible, because bits depends on attacker power and value as well as hero upgrades)

**Comment by Urban-Hacker** [comment]
Maybe it can be like in the game Halo 3. IF you enable a handicap it forces it on you, eg: you cannot do upgrade etc.

Maybe this could be only on a few selected level that we know are possible to achieve

**Comment by Cwpute** [comment]
If it's forced on you that would require more work to implement i believe... even more so to manually check every level for handicap feasability 😬 

----- Mail original -----

De: "Urban-Hacker" ***@***.***> 
À: "ochadenas/cpudefense" ***@***.***> 
Cc: "Cwpute" ***@***.***>, "Comment" ***@***.***> 
Envoyé: Mardi 19 Mars 2024 22:51:29 
Objet: Re: [ochadenas/cpudefense] Should we increase price / difficulty / whatever? (beginner vs expert player) (Discussion #130) 



Maybe it can be like in the game Halo 3. IF you enable a handicap it forces it on you, eg: you cannot do upgrade etc. 
Maybe this could be only on a few selected level that we know are possible to achieve

**Comment by ochadenas** [comment]
Sounds interesting, but quite complex. For the moment at least, I've implemented a simple solution in 1.34: There is a new hero, Leibniz, that becomes available after Grace Hopper and allows for upgrades of MEM chips. And the upgrade prices are now 36 for the 1st upgrade, 90 for the second and 225 for the 3rd. The chip is limited to 4 slots max, but that can easily be changed.

**Comment by ochadenas** [comment]
The most straight forward thing is probably what you've suggested some time ago already: Have a second, separate set of cryptocoins/heroes, so that you start at zero on the endless series. The general difficulty would have to be reduced, of course.

This would have the benefit of decoupling the difficulty levels and make tuning the difficulty easier. I'm just a bit scared of implementing this, it sounds like quite a change in mechanics, and errors are likely to happen.

I don't know Halo 3. What I have in mind is "Rogue Adventure", an Android game I like very much. There they have introduced an endless series where for each set of levels, a certain (and changing) handicap is applied. Sometimes you get the choice either to disable a malus or to have an additional benefit. This could be a model. And still decide later whether an additional "campaign" mode makes sense or whether the "campaign" ideas can be integrated into the endless levels.

Btw., another common point with Rogue Adventure is that the "endless" mode is quite nice there, but does not add as much thrill as you should think, and that it is obviously extremely hard to get the balance right. I have climbed up to level 385 easily, and in the end this was rather boring.

**Reply by Cwpute** [reply]
> There they have introduced an endless series where for each set of levels, a certain (and changing) handicap is applied. Sometimes you get the choice either to disable a malus or to have an additional benefit. 

That sounds awesome!

**Comment by Urban-Hacker** [comment]
@ochadenas Halo 3 is on the surface very different from CPU defense or tower defense in general, its a first person shooter similar to Doom / Quake / Etc.

But it as a special game mode in which you can replay any mission with some kinda of handicap. For instance enemies have only one weapon: grenade, or you do not have GUI, so you play just on visual cues. Some of them are just for fun: eg, every time you shoot a foe you get fireworks.

The way you unlock them is by exploring levels and collecting "skulls" this adds a double re-playability. First you need to finish the game and then replay some levels in order to collect the skulls. Then those skulls can be used to re-play levels with various handicap / gameplay changes.

Any of those skulls can be enabled / disabled when you replay a level.

I think initially before doing any changes to CPU defense, achievement should be implemented. This enables to have secondary / fun objectives which then offer the base to which the rest can be extended.

**Comment by ochadenas** [comment]
I have finally done the changes necessary to have a separate set of coins and heroes for the endless levels. Starting from 1.34; if I do not find any major bugs, I'll try to publish it on github next sunday.

Of course, there will still be bugs ...

The change means for a "standard" player that after finishing the "basic" series (normal + turbo), they will start the "endless" series without coins and heroes. To compensate for it, the difficulty has been heavily reduced. Well, at least I have reached level 12 during my first tests, and then it starts becoming a challenge.

For the experienced player, the game will transfer the coins and heroes to the new system as best as it can. It distributes the coins got so far fairly between the "basic" and the "endless" series. If the cost of current heroes exceeds the available coins, the heroes are reset, and the player must hire them once again. Which means that, more or less, you will have to buy the whole set of heroes another time. This should limit inflation a bit.

Because these are quite a lot of changes, I'd rather stop here for the moment and keep all the other ideas for later.
